test/crau_verify.rs

```diff
@@ -28,15 +42,34 @@ fn main() -> Result<(), Box<dyn Error>> {
 
     // Read update payload from srcpath, read delta update header from the payload.
     let upfile = fs::File::open(srcpath.clone())?;
-    let header = delta_update::read_delta_update_header(&upfile)?;
+
+    let freader = &mut BufReader::new(upfile);
+    let header = delta_update::read_delta_update_header(freader)?;
+
+    let mut delta_archive_manifest: proto::DeltaArchiveManifest = Default::default();
 
     // Extract signature from header.
-    let sigbytes = delta_update::get_signatures_bytes(&upfile, &header)?;
+    let sigbytes = delta_update::get_signatures_bytes(freader, &header, &mut delta_archive_manifest)?;
+
+    // Parse signature data from the signature containing data, version, special fields.
+    let tmpdir = Path::new("/tmp/ue_rs");
```

Comment:
`/tmp` is backed by RAM and we would fill the RAM with a large dev-container sysext. I think the previous approach of strong things in subdirectories is required for the large data parts unless we do a streaming processing.

Reply:
Ah, now I see, this is just for the test, that's a different problem: Tests can run in parallel and if you use the same path they will corrupt their data, you need a random path.